### PR TITLE
rados: allow "rados purge" to delete objects when osd is full

### PR DESCRIPTION
**Pull request by liupan1111**
Signed-off-by: Pan Liu <liupan1111@gmail.com>

### CODE REVIEW

**Comment by liupan1111** [comment]
@ceph-jenkins retest this please.

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -2855,6 +2855,7 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       goto out;
     }
     io_ctx.set_namespace(all_nspaces);
+	io_ctx.set_osdmap_full_try();
```
Comment (dillaman) [inline src/tools/rados/rados.cc]: Nit: indentation looks incorrect

Comment (liupan1111) [inline src/tools/rados/rados.cc]: @dillaman done, thanks.

File: src/tools/rados/rados.cc [hunk 2]
```diff
@@ -2855,6 +2855,7 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       goto out;
     }
     io_ctx.set_namespace(all_nspaces);
+    io_ctx.set_osdmap_full_try();
```
Comment (dzafman) [inline src/tools/rados/rados.cc]: I think we should consider supporting the --force-full option just like the "rm" object command does.

Comment (dzafman) [inline src/tools/rados/rados.cc]: Unfortunately, from  what I can see using aio_remove() can only use set_osdmap_full_try(), so for now I withdraw my suggestion to support --force-full unless/until another option is added to set that in the io_ctx.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-03-16_11:38:52-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2017-03-17_04:29:25-rados-wip-kefu-testing---basic-mira/

failed tests tracked by http://tracker.ceph.com/issues/19312

**Pull request merged**